tests/integration/conftest.py

```diff
@@ -10,3 +11,19 @@ async def loki_charm(ops_test: OpsTest):
     """Loki charm used for integration testing."""
     charm = await ops_test.build_charm(".")
     return charm
+
+
+@pytest.fixture(scope="module", autouse=True)
+def copy_loki_library_into_tester_charm(ops_test):
+    """Ensure that the tester charm uses the current Prometheus library."""
+    library_path = "lib/charms/loki_k8s/v0/loki_push_api.py"
+    install_path = "tests/integration/loki-tester/" + library_path
+    shutil.copyfile(library_path, install_path)
```

Comment:
I ain’t no @sed-i, but wouldn’t it be possible to simply have the tester charm lib be a relative symlink to the root-level lib folder. Done it this way, it looks plenty brittle.

Reply:
Have not tried it. But certainly worth a try. It would work if charmcraft resolves the symlink into an actual file and copies the file (rather than a symlink) into the built charm when packing. Of the top of my head I do not know if charmcraft actually does that.